libraries/app/CMakeLists.txt

```diff
@@ -16,11 +16,12 @@ add_library( graphene_app
 # need to link graphene_debug_witness because plugins aren't sufficiently isolated #246
 target_link_libraries( graphene_app
                        graphene_market_history graphene_account_history graphene_elasticsearch graphene_grouped_orders
-                       graphene_api_helper_indexes
+                       graphene_api_helper_indexes ${QUERY_TXID} 
                        graphene_chain fc graphene_db graphene_net graphene_utilities graphene_debug_witness )
 target_include_directories( graphene_app
                             PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}/include"
-                            "${CMAKE_CURRENT_SOURCE_DIR}/../egenesis/include" )
+                            "${CMAKE_CURRENT_SOURCE_DIR}/../egenesis/include" 
+                            "${CMAKE_CURRENT_SOURCE_DIR}/../plugins/query_txid_object/include")
```

Comment:
I think this is superfluous. The library dependency above should make the plugin include files available to app.

Reply:
Now I get it. You need the include even when the plugin is disabled, because it declares the structure that is used in the return value of the API calls.

You can work around this by `typedef`ing `query_trx_info` to some other known structure in that case (the API call should always throw an error, so the actual return type will not be used).